packages/purpose-process/test/updatePurpose.test.ts

```diff
@@ -260,6 +270,325 @@ describe("updatePurpose and updateReversePurpose", () => {
+      messageType: DraftPurposeUpdatedV2,
+      payload: writtenEvent.data,
+    });
+
+    const expectedPurpose: Purpose = createUpdatedPurpose(
+      purposeForDeliver,
+      updateContentWithoutTitle,
+      validRiskAnalysis,
+      writtenPayload.purpose!.riskAnalysisForm!
+    );
+
+    expect(writtenPayload.purpose).toEqual(toPurposeV2(expectedPurpose));
+    expect(writtenPayload.purpose).toEqual(toPurposeV2(purpose));
+    expect(isRiskAnalysisValid).toBe(true);
+  });
+  it("should succeed when requester is Consumer Delegate and the Purpose is in a updatable state and the e-service is in mode RECEIVE", async () => {
+    const authData = getRandomAuthData();
+
+    const delegation = getMockDelegation({
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: purposeForReceive.eserviceId,
+      delegatorId: purposeForReceive.consumerId,
+      delegateId: authData.organizationId,
+      state: delegationState.active,
+    });
+
+    await addOnePurpose(purposeForReceive);
+    await addOneDelegation(delegation);
+    await addSomeRandomDelegations(purposeForDeliver, addOneDelegation);
+    await writeInReadmodel(toReadModelEService(eServiceReceive), eservices);
+    await writeInReadmodel(toReadModelTenant(tenant), tenants);
+
+    const { purpose, isRiskAnalysisValid } =
+      await purposeService.updateReversePurpose(
+        purposeForReceive.id,
+        reversePurposeUpdateContent,
+        {
+          authData,
+          correlationId: generateId(),
+          logger: genericLogger,
+          serviceName: "",
+        }
+      );
+
+    const writtenEvent = await readLastPurposeEvent(purposeForReceive.id);
+    expect(writtenEvent).toMatchObject({
+      stream_id: purposeForReceive.id,
+      version: "1",
+      type: "DraftPurposeUpdated",
+      event_version: 2,
+    });
+
+    const writtenPayload = decodeProtobufPayload({
+      messageType: DraftPurposeUpdatedV2,
+      payload: writtenEvent.data,
+    });
+
+    const expectedPurpose: Purpose = createUpdatedPurpose(
+      purposeForReceive,
+      reversePurposeUpdateContent,
+      validRiskAnalysis,
+      writtenPayload.purpose!.riskAnalysisForm!
+    );
+
+    expect(writtenPayload.purpose).toEqual(toPurposeV2(expectedPurpose));
+    expect(writtenPayload.purpose).toEqual(toPurposeV2(purpose));
+    expect(isRiskAnalysisValid).toBe(true);
+  });
+  it("should succeed when requester is Consumer Delegate and the eservice was created by a delegated tenant and the Purpose is in a updatable state and the e-service is in mode DELIVER", async () => {
+    const producerDelegator = {
+      ...getMockTenant(),
+      id: generateId<TenantId>(),
+      kind: tenantType,
+    };
+    const producer = {
+      ...getMockTenant(),
+      id: generateId<TenantId>(),
+      kind: tenantType,
+    };
+    const consumerDelegator = {
+      ...getMockTenant(),
+      id: generateId<TenantId>(),
+      kind: tenantType,
+    };
+    const consumer = {
```

Comment:
```suggestion
    const producer= {
      ...getMockTenant(),
      id: generateId<TenantId>(),
      kind: tenantType,
    };
    const producerDelegate = {
      ...getMockTenant(),
      id: generateId<TenantId>(),
      kind: tenantType,
    };
    const consumer = {
      ...getMockTenant(),
      id: generateId<TenantId>(),
      kind: tenantType,
    };
    const consumerDelegate = {
```
I suggest to rename these because the real consumer and producer are the delegators, so it's clearer to name them as such and specify the delegates